packages/jest-snapshot/src/InlineSnapshots.ts

```diff
@@ -6,109 +6,130 @@
-  }
-
-  // Custom parser API was added in 1.5.0
-  if (semver.lt(prettier.version, '1.5.0')) {
-    throw new Error(
-      `Jest: Inline Snapshots require prettier>=1.5.0.\n` +
-        `Please upgrade "prettier".`,
-    );
-  }
+  // TODO same as above
+  const prettier = prettierPath
+    ? (require(require.resolve(prettierPath, {
+        [Symbol.for('OUTSIDE_JEST_VM_RESOLVE_OPTION')]: true,
+      } as any)) as typeof import('prettier'))
+    : null;
 
   const snapshotsByFile = groupSnapshotsByFile(snapshots);
 
   for (const sourceFilePath of Object.keys(snapshotsByFile)) {
     saveSnapshotsForFile(
       snapshotsByFile[sourceFilePath],
       sourceFilePath,
-      prettier,
-      babelTraverse,
+      prettier && semver.gte(prettier.version, '1.5.0') ? prettier : null,
     );
   }
 }
 
 const saveSnapshotsForFile = (
   snapshots: Array<InlineSnapshot>,
   sourceFilePath: Config.Path,
-  prettier: Prettier,
-  babelTraverse: BabelTraverse,
+  prettier: any,
 ) => {
   const sourceFile = fs.readFileSync(sourceFilePath, 'utf8');
 
-  // Resolve project configuration.
-  // For older versions of Prettier, do not load configuration.
-  const config = prettier.resolveConfig
-    ? prettier.resolveConfig.sync(sourceFilePath, {editorconfig: true})
-    : null;
-
-  // Detect the parser for the test file.
-  // For older versions of Prettier, fallback to a simple parser detection.
-  // @ts-expect-error
-  const inferredParser: PrettierParserName = prettier.getFileInfo
-    ? prettier.getFileInfo.sync(sourceFilePath).inferredParser
-    : (config && config.parser) || simpleDetectParser(sourceFilePath);
+  // TypeScript projects may not have a babel config; make sure they can be parsed anyway.
+  const presets = [require.resolve('babel-preset-current-node-syntax')];
+  const plugins: Array<PluginItem> = [];
+  if (/\.tsx?$/.test(sourceFilePath)) {
+    plugins.push([
+      require.resolve('@babel/plugin-syntax-typescript'),
+      {isTSX: sourceFilePath.endsWith('x')},
+      // unique name to make sure Babel does not complain about a possible duplicate plugin.
+      'TypeScript syntax plugin added by Jest snapshot',
+    ]);
+  }
 
-  // Record the matcher names seen in insertion parser and pass them down one
+  // Record the matcher names seen during traversal and pass them down one
   // by one to formatting parser.
   const snapshotMatcherNames: Array<string> = [];
 
-  // Insert snapshots using the custom parser API. After insertion, the code is
-  // formatted, except snapshot indentation. Snapshots cannot be formatted until
-  // after the initial format because we don't know where the call expression
-  // will be placed (specifically its indentation).
-  const newSourceFile = prettier.format(sourceFile, {
-    ...config,
-    filepath: sourceFilePath,
-    parser: createInsertionParser(
-      snapshots,
-      snapshotMatcherNames,
-      inferredParser,
-      babelTraverse,
-    ),
+  const ast = parseSync(sourceFile, {
```

Comment:
should we make this async? breaking, but I think it should be possible? maybe not

Reply:
Currently seems like it wouldn't have any perf gains since this is per worker and not parallelizable with anything else. Leave as a future possibility I'd say, this change is also already big enough